app/src/main/java/com/tip/lunchbox/utilities/FloatingButtonBehaviour.java

```diff
@@ -0,0 +1,53 @@
+package com.tip.lunchbox.utilities;
+
+import android.content.Context;
+import android.util.AttributeSet;
+import android.util.Log;
+import android.view.View;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.coordinatorlayout.widget.CoordinatorLayout;
+
+import com.google.android.material.floatingactionbutton.FloatingActionButton;
+import com.tip.lunchbox.R;
+
+/**
+ * Custom behaviour class for the 'Clear Filter' chip to interact with the Bottom Sheet within the
+ * CoordinatorLayout. Does not replicate the FloatingActionButton anchoring behaviour, rather
+ * designed for the chip to hide below the Bottom Sheet on pull up.
+ */
+public class FloatingButtonBehaviour extends CoordinatorLayout.Behavior<View> {
```

Comment:
Just curious, wouldn't simply using the an anchor and setting the visibility as `GONE` for the clear filter chip been easier?

Reply:
Well I felt anchoring didn't look too good with the the bottom sheet, because the chip gets pulled up along with it, and it doesn't really suit the purpose (at least in my opinion)

Reply:
I was actually wondering if you can set visibility as `GONE` inside the `onSlide` function of `bottomSheetBehavior.setBottomSheetCallback`

Reply:
I can try, but just so I'm clear, you're suggesting anchoring to the bottom sheet and setting the visibility to GONE in the onSlide function?

Reply:
Yeah, I believe this will also work. Then you can again set visibility as VISIBLE once in collapsed state

Reply:
I will try it out 👍🏻 although I spent so much time on that custom behaviour xD

Reply:
Okay, so we will keep this approach only 😆 
Just for learning/confirming may be you can try that approach also

Reply:
Sure, anything works. I can test out the other approach within some time.